File: julian_date/ruby_julian_date/julian_date.rb
#!/usr/bin/ruby -w
require 'date'
#comments

def julian_date( y, m, d )
    y += 8000
    if m < 3 then
      y = y - 1
      m = m + 12
    end

    return ((y * 365) + (y / 4) - (y / 100) + (y / 400) - 1200820 + (m * 153 + 3) / 5 - 92 + d - 1) - 0.5
end

def main()
  if ARGV.length != 0 then
    puts "Usage: #{$0} <noargs>"
    exit(1)
  end
  puts julian_date(Date.today.year, Date.today.month, Date.today.day)
end

main()

File: julian_date/perl_julian_date/julian_date.pl
#!/usr/bin/perl -w

use strict;
use warnings;
use Time::localtime;
use Date::Calc qw();

#does not work yet
sub julian_date {
  my $y = shift;
  my $m = shift;
  my $d = shift;

  $y += 8000;
  if( $m < 3 ) {
    $y--;
    $m += 12;
  }
  return int(($y * 365) + ($y / 4) - ($y / 100) + ($y / 400) - 1200820 + ($m * 153 + 3) / 5 - 92 + $d - 1) - 0.5;
}

sub main {
  my ($year, $month, $mday) = Date::Calc::Today();

  if( $#ARGV+1 != 0 ) {
    print STDERR "Usage: $0 <noargs>\n";
    exit(1);
  }
  print &julian_date($year, $month, $mday), "\n";
  exit(0);
}

&main();
